frappe/core/doctype/comment/comment.py

```diff
@@ -102,6 +122,91 @@ def remove_comment_from_cache(self):
 
 		update_comments_in_parent(self.reference_doctype, self.reference_name, _comments)
 
+	def validate_spam(self):
+		from bs4 import BeautifulSoup
+
+		spam_filtering = frappe.db.get_single_value("Akismet Settings", "spam_filtering")
+		if self.comment_type != "Comment":
+			return
```

Comment:
This is also only required for guest comments. We can skip users on site to avoid extra API calls. (Also tons of comments are made automatically by code/automations, no need to validate those)